File: votevoice-server/VoteVoice/PollService/Models/Poll.cs
using PollService.Business.Entities;
using System.ComponentModel.DataAnnotations;

namespace PollService.Models
{
    public class Poll : BaseEntity, IPoll
    {
        [Key]
        public long PollId { get; set; }

        public long UserId { get; set; } // Reference to User Service

        [Required]
        public string Title { get; set; }

        public string Description { get; set; }

        public bool IsActive { get; set; } = true;

        public DateTime ExpiryDate { get; set; }
    }

}

File: votevoice-client/vote-voice/src/app/interfaces/poll.ts
import { PollOption } from './poll-option';
export class Poll {
    pollId!: number;
    userId!: number;
    title!: string;
    description!: number;
    isActive!: boolean;
    firstName!: string;
    lastName!: string;
    profileImage!: any;
    expiryDate!: Date;
    totalVotes!: number;
    pollOptions!: PollOption[];
}